### PR TITLE
 Invalidate cache on form completion

### LINKED ISSUE #2234: Invalidation/refresh cache after form completion
**Describe the feature request.**
Since we currently cache the registers and profiles data we need to refresh/invalidate the cache every time a patient or family is removed from the app.

For the family register we need to filter out `inactive` families. The [Filter out inactive families using custom search parameters](https://github.com/opensrp/fhircore/issues/2233) addresses this.

**Acceptance criteria**
- [ ] When a family is removed, it should not appear on the household register
- [ ] When a patient is removed, it should not appear on the patient register

**Area path**
A list of ordered steps in the app on usage of the feature to support anyone testing it e.g. Code reviewer, QA e.g.
1. Navigate to HouseHold register
2. Click on one family
3. Click on `Remove Family` on the household profile overflow menu
4. Fill out and save the `Remove Family` questionnaire
5. Navigate back to the `HouseHold` register
6. The removed family should not appear on the register

**Implementation plan (For Engineers)**
- [ ] Add a `refreshContent` boolean that determines whether the register cache should be invalidated
- [ ] Add a `dataRefreshLivedata` boolean live data object in `AppMainViewModel`
- [ ] Check whether `questionnaireConfig.refreshContent` is true in `AppMainActivity.onSubmitQuestionnaire`. If true post a value of true using `dataRefreshLivedata`
- [ ] Observer `dataRefreshLivedata` in `RegisterFragment.onViewCreated`
- [ ] When `dataRefreshLivedata` is updated to `true` invoke `registerViewModel.retrieveRegisterUiState` with a `clearCache` value of `true`
- [ ] Reset the `dataRefreshLivedata` value to false

### PR DESCRIPTION
**Pull request by Rkareko**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2234

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 

**Code Reviewer Checklist**
- [x] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [x] I have verified any strings visible on UI components are in the `strings.xml` file
- [x] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [x] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [x] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/RegisterRepository.kt [hunk 1]
```diff
@@ -308,6 +308,10 @@ constructor(
         if (resourceType == ResourceType.Patient) {
           filter(TokenClientParam(ACTIVE), { value = of(true) })
         }
+        // For Group return only active families count
```
Comment (ellykits) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/RegisterRepository.kt:311]:
```suggestion
        // Filter active Groups
```

**Review by ellykits — DISMISSED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt [hunk 2]
```diff
@@ -338,6 +336,10 @@ constructor(
     }
   }
 
+  fun dataRefreshLivedataPostValue(refresh: Boolean) {
+    dataRefreshLivedata.postValue(refresh)
+  }
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt:341]: why add this?

Comment (Rkareko) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainViewModel.kt]: This was for testing purposes.
The code and test have been updated.

**Review by pld — COMMENTED** [review]
lgtm!

**Review by pld — APPROVED** [review]

**Pull request merged**